crates/typst-macros/src/elem.rs

```diff
@@ -480,88 +472,904 @@ fn create_param_info(field: &Field) -> TokenStream {
+                if self.span().is_detached() {
+                    self.span = span;
+                }
+            }
+
+            fn location(&self) -> Option<#model::Location> {
+                #location
+            }
+
+            fn set_location(&mut self, location: #model::Location) {
+                #set_location
+            }
+
+            fn label(&self) -> Option<#model::Label> {
+                #label
+            }
+
+            fn set_label(&mut self, label: #model::Label) {
+                #set_label
+            }
+
+            fn push_guard(&mut self, guard: #model::Guard) {
+                self.guards.push(guard);
+            }
+
+            fn is_guarded(&self, guard: #model::Guard) -> bool {
+                self.guards.contains(&guard)
+            }
+
+            fn is_pristine(&self) -> bool {
+                self.guards.is_empty()
+            }
+
+            fn mark_prepared(&mut self) {
+                #mark_prepared
+            }
+
+            fn needs_preparation(&self) -> bool {
+                #needs_preparation
+            }
+
+            fn is_prepared(&self) -> bool {
+                #prepared
+            }
+
+            fn dyn_hash(&self, mut hasher: &mut dyn ::std::hash::Hasher) {
+                <Self as ::std::hash::Hash>::hash(self, &mut hasher);
+            }
+
+            fn dyn_eq(&self, other: &#model::Content) -> bool {
+                if let Some(other) = Self::unpack(other) {
+                    <Self as ::std::cmp::PartialEq>::eq(self, other)
+                } else {
+                    false
+                }
+            }
+
+            fn dyn_clone(&self) -> ::std::sync::Arc<dyn #model::NativeElement> {
+                ::std::sync::Arc::new(Clone::clone(self))
+            }
+
+            fn field(&self, id: u8) -> Option<::typst::eval::Value> {
+                let id = <#enum_ident as ::std::convert::TryFrom<u8>>::try_from(id).ok()?;
+                match id {
+                    #enum_ident::Label => #label_field,
+                    #(
+                        #field_matches
+                    )*
+                    _ => None,
+                }
+            }
+
+            fn fields(&self) -> Dict {
+                let mut fields = Dict::new();
+                #(#field_dict)*
+                #(#field_opt_dict)*
+                fields
+            }
+
+            fn set_field(&mut self, id: u8, value: Value) -> ::typst::diag::StrResult<()> {
+                let id = <#enum_ident as ::std::convert::TryFrom<u8>>::try_from(id)
+                    .map_err(|_| ::ecow::eco_format!(#unknown_field, id))?;
+                match id {
+                    #(
+                        #field_set_matches
+                    )*
+                    #(
+                        #field_inherent_matches
+                    )*
+                    #(
+                        #field_not_set_matches
+                    )*
```

Comment:
```suggestion
                    #(#field_set_matches)*
                    #(#field_inherent_matches)*
                    #(#field_not_set_matches)*
```
and the same in similar places